projects/libpng-proto/png_proto_mutator.cc

```diff
@@ -0,0 +1,34 @@
+#include "libprotobuf-mutator/src/libfuzzer/libfuzzer_macro.h"
+#include "png_fuzz_proto.pb.h"
+
+struct PngProtoMutator
+{
+	PngProtoMutator(
+		google::protobuf::Descriptor const* _desc,
+		std::function<void(google::protobuf::Message*, unsigned int)> _callback
+	)
+	{
+		protobuf_mutator::libfuzzer::RegisterPostProcessor(_desc, _callback);
+	}
+};
+
+/// Custom mutation: Otherchunk unknown_type -> known_type
+static PngProtoMutator addCustomChunk(
+	OtherChunk::descriptor(),
+	[](google::protobuf::Message* _message, unsigned int _seed)
+	{
+		OtherChunk* otherChunk = static_cast<OtherChunk*>(_message);
+		// Mutate with a probability of roughly 1/47
+		// 47 has been chosen ad-hoc
+		if (_seed % 47 == 0)
+		{
+			// If otherChunk is unknown type, mutate
+			// it to known type
+			if (otherChunk->has_unknown_type())
+			{
+				// This is our custom mutation
+				otherChunk->set_known_type(_seed);
```

Comment:
it's essentially
set_known_type(n * 47)

Reply:
What is `n` in your comment?

The actual value passed to `set_known_type` should not matter since the proto converter is using a modulo on the size of known chunks to decide which one to pick.

https://github.com/google/oss-fuzz/blob/9532d4781a8a7539591f6edc0b86e82786fffafa/projects/libpng-proto/png_proto_fuzzer_example.cc#L90

Reply:
Ah, I got it now. You meant all set known types will be multiples of 47.

I chose 47 because it is a prime and it is large enough to not apply the said mutation too often.

I'm thinking if multiples of 47 mod number of known chunks distribute roughly uniformly across the set of known chunks, but perhaps I'm over thinking it.

Reply:
yes: 0 47,94...
you can try: otherChunk->set_known_type(_seed/47);